libfwupd/fwupd-client.c

```diff
@@ -344,11 +344,23 @@ fwupd_client_set_host_product(FwupdClient *self, const gchar *host_product)
 	fwupd_client_object_notify(self, "host-product");
 }
 
-static void
+/**
+ * fwupd_client_set_host_machine_id:
+ * @self: a #FwupdClient
+ * @host_machine_id: the host machine ID, usually a SHA-256 hash
+ *
+ * Manually sets the machine ID. This is normally only required for self tests.
```

Comment:
I wonder if it's worth using mock instead of adding new code like this.

Reply:
Actually, I've reverted this -- and just used a conditional so that the machine_id can be unset.